sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -635,6 +642,122 @@ class SparkSession private(
 
 object SparkSession {
 
+  /**
+   * Builder for [[SparkSession]].
+   */
+  class Builder {
+
+    private[this] val options = new scala.collection.mutable.HashMap[String, String]
```

Comment:
But, forget about my comments. `Builder` is so simple and current implementation is solid, too.